doc/user_guide/04_core_methods.md

```diff
@@ -373,3 +267,256 @@ plt.xlabel("Time (s)")
+```{code-cell} ipython3
+metadata = pd.DataFrame(
+    index=group.keys(),
+    data=["pfc", "ofc", "hpc"],
+    columns=["region"]
+)
+
+tsgroup = nap.TsGroup(group, metadata=metadata)
+print(tsgroup)
+```
+
+
+#### `IntervalSet`
+Metadata added to `IntervalSet` must match the number of intervals, or the length of its `index` property. 
+
+```{code-cell} ipython3
+metadata = {
+    "reward": [1, 0, 1],
+    "choice": ["left", "right", "left"],    
+}
+
+intervalset = nap.IntervalSet(starts, ends, metadata=metadata)
+print(intervalset)
+```
+
+Metadata can be initialized as a DataFrame using the metadata argument, or it can be inferred when initializing an `IntervalSet` with a DataFrame.
+```{code-cell} ipython3
+df = pd.DataFrame(
+    data=[[0, 5, 1, "left"], [10, 15, 0, "right"], [20, 25, 1, "left"]], 
+    columns=["start", "end", "reward", "choice"]
+    )
+
+intervalset = nap.IntervalSet(df)
+print(intervalset)
+```
+
+#### `TsdFrame`
+Metadata added to `TsdFrame` must match the number of data columns, or the length of its `columns` property. 
+```{code-cell} ipython3
+metadata = {
+    "color": ["red", "blue", "green"], 
+    "position": [10,20,30]
+    }
+
+tsdframe = nap.TsdFrame(d=d, t=t, columns=["a", "b", "c"], metadata=metadata)
+print(tsdframe)
+```
+
+When initializing with a DataFrame, the DataFrame index must match the `TsdFrame` columns.
+```{code-cell} ipython3
+metadata = pd.DataFrame(
+    index=["a", "b", "c"],
+    data=[["red", 10], ["blue", 20], ["green", 30]], 
+    columns=["color", "position"],
+)
+
+tsdframe = nap.TsdFrame(d=d, t=t, columns=["a", "b", "c"], metadata=metadata)
+print(tsdframe)
+```
+
+#### `set_info`
+After creation, metadata can be added using the class method `set_info()`. Metadata can be passed as a dictionary or pandas DataFrame as the first positional argument, or metadata can be passed as name-value keyword arguments.
+
+```{admonition} Note
+The remaining metadata examples will be shown on a `TsGroup` object; however, all examples can be directly applied to `IntervalSet` and `TsdFrame` objects.
+```
+
+```{code-cell} ipython3
+tsgroup.set_info(unit_type = ["multi", "single", "single"])
+print(tsgroup)
+```
+
+### Accessing metadata
+Metadata is stored as a pandas DataFrame, which can be previewed using the `metadata` attribute.
+
+```{code-cell} ipython3
+print(tsgroup.metadata)
+```
+
+Single metadata columns or lists of columns can be retrieved using the `get_info()` class method:
+```{code-cell} ipython3
+print(tsgroup.get_info("region"))
+```
+
+### Metadata properties
+User-set metadata is mutable and can be overwritten.
+```{code-cell} ipython3
+tsgroup.set_info(region=["A","B","C"])
+print(tsgroup.get_info("region"))
+```
+
+If the metadata name does not overlap with an existing class column, it can be set and accessed via key indexing (i.e. using square brackets).
```

Comment:
Shouldn't this be in the section Accessing metadata?

Reply:
Also why mentioning class column only?

Reply:
I changed this to explicitly mention the exceptions for IntervalSet's "start" and "end"